server/manager/src/main/java/org/apache/accumulo/manager/Manager.java

```diff
@@ -1120,6 +1122,9 @@ boolean canSuspendTablets() {
       if (null != upgradeMetadataFuture) {
         upgradeMetadataFuture.get();
       }
+      if (null != upgradeFilesFuture) {
+        upgradeFilesFuture.get();
+      }
```

Comment:
Correct, it doesn't matter in this case. But I did have the same concerns about future problems with the upgrader. Since I didn't have a concrete example, I wasn't sure when `upgradeFiles()` should get called. I ended up settling with this concurrent upgrade since that works for the current changes. 

Any thoughts on where in the upgrade order the files would ideally get upgraded? I was originally thinking last, after the metadata. Maybe the files should be done first before anything (but after ZK?) so we know what paths to store? Like I said, I couldn't think of any specific upgrade example where files were dependent on other changes.

Reply:
Well, usually we don't upgrade files at all, because the number of files tends to be excessive and not scalable. So, we support upgrading files over time in a distributed way, by reading multiple older versions, and writing only newer versions. See RFile code. I think we even still have some remaining code for reading MapFiles (which we used before RFile existed). In general, we don't want to upgrade files at all at this phase, certainly not isolated to a single machine, when we could have thousands and thousands of files across HDFS.

In this case, I was concerned that these files being dropped depend on metadata changes... since there could be references to these sorted WAL files in the metadata. That would be one example. In general, though, we just don't upgrade files at this point at all.

Reply:
Since we don't upgrade files, should this fix be made somewhere else? I didn't want to clutter the recovery code for many versions to handle deleting the files one time.

Reply:
I think it's probably okay to upgrade the files at manager startup through this mechanism. The part I'm concerned about is changing the upgrade interface to provide a general mechanism to upgrade files, as that could encourage misuse. Maybe we can just consider these part of the "metadata" and perform the upgrade there instead of add the new "upgradeFiles" stuff?

Reply:
I reverted the changes I made adding a new `upgradeFiles` method to the Upgrade interface and now this change is much simpler.